notify/sns/sns.go

```diff
@@ -198,6 +241,38 @@ func validateAndTruncateMessage(message string, maxMessageSizeInBytes int) (stri
 	return string(truncated), true, nil
 }
 
+func validateAndTruncateSubject(logger log.Logger, subject string, modifiedReasons *[]string) string {
+	if subject == "" {
+		*modifiedReasons = append(*modifiedReasons, fmt.Sprintf(ComponentAndModifiedReason, Subject, SubjectEmpty))
+		level.Warn(logger).Log("msg", "Subject has been modified because it is empty.", "originalSubject", subject)
+		return SubjectEmpty
+	}
+
+	if !isASCIINonControl(subject) {
+		*modifiedReasons = append(*modifiedReasons, fmt.Sprintf(ComponentAndModifiedReason, Subject, SubjectContainsIllegalChars))
+		level.Warn(logger).Log("msg", "Subject has been modified because it contains control- or non-ASCII characters.", "originalSubject", subject)
```

Comment:
```suggestion
		level.Warn(logger).Log("msg", "subject has been modified because it contains control- or non-ASCII characters", "subject", subject)
```